### ISSUE TITLE
Error `code: 0. path parameter is invalid.  It should begin with m or an integer number.`

### ISSUE THREAD

**Issue by tkalfigo** [dan-da/hd-wallet-derive#7]
This error is thrown with following paths:

Case 1
```
$ hd-wallet-derive -g --coin=BTC-testnet --key=tpubDGYAReSwceYYJz5cFBPK5uNP8YsHqmz7tCqEc7bCFwmo6Z89YABkuMLC1LkVKs5SP7jVCf4sPwaJraBewJUE6gPr1riYncUYUBeqVFaEL63 --path="0/"
```
Case 2

```
$ hd-wallet-derive -g --coin=BTC-testnet --key=tpubDGYAReSwceYYJz5cFBPK5uNP8YsHqmz7tCqEc7bCFwmo6Z89YABkuMLC1LkVKs5SP7jVCf4sPwaJraBewJUE6gPr1riYncUYUBeqVFaEL63 --path="0/0"
```

In both cases the condition in the error message is fulfilled. The path starts with an integer.

Also these paths should/could be valid.

In particular the 1st case is handled nicely by [bip32.org](http://bip32.org) where if the path ends with `/` it gives 1st index keys i.e. defaults to `/0`.

Regarding the 2nd case I'd expect `hd-wallet-derive` to not throw error, given how it currently handles the following cases:
*  `--path=""` and `--path="m"` are the same and generate first level i.e. `m/0`, `m/1`...
*  `--path="0"` and `--path="m/0"` are the same and generate 2nd level i.e. `m/0/0`, `m/0/1`...

Does it not make sense for `--path="0/0"` to generate 3rd level? i.e. `m/0/0/0`, `m/0/0/1`...

**Comment by dan-da** [comment]
hi, I will look into the path issue(s) you raise later as it's late here.  

First though, your command would not work because it should be BTC-test, not testnet.   (The naming [main/test/regtest] comes from the naming of the classes in chainparams.cpp.)  I might alias testnet in a future release.

**Comment by tkalfigo** [comment]
Good catch! I renamed it from my local exmaples that used `BTC-regtest` but that doesn't really change the claim.

**Comment by dan-da** [comment]
path validation is improved/fixed and tested in c85c0d160747d615b1c767a233a5563fa091a94f.  Additional check for 0/0 in 2c6e3ed278e3f9726f130ad109d38f1936edc257.   Closing.

**Comment by dan-da** [comment]
related: https://github.com/Bit-Wasp/bitcoin-php/issues/715
